lib/storage/metadata/mongoclient/MongoClientInterface.js

```diff
@@ -451,12 +459,19 @@ class MongoClientInterface {
         // eslint-disable-next-line
         objVal.versionId = params.versionId;
         const vObjName = formatVersionKey(objName, params.versionId);
-        c.findOne({ _id: objName }, (err, checkObj) => {
+        const query = { _id: objName };
+        if (params.condPut && params.condPut.tag) {
+            query.tag = params.condPut.tag;
+        }
+        c.findOne(query, (err, checkObj) => {
             if (err) {
                 log.error('putObjectVerCase3: mongoDB error finding object');
                 return cb(errors.InternalError);
             }
             const objUpsert = !checkObj;
+            if (objUpsert && params.condPut.tag) {
```

Comment:
I'm a bit confused about the logic here, can you in a brief code comment explain it? What I understand is, if `findOne` does not return a document (but why `err` is null then), `objUpsert` is true (so we ask for an insert if the object does not exist) but if we asked for a tag check then we don't apply the update (but then, why not returning the `versionId` like done at line 522?)